src/lib/mollie/api/client.coffee

```diff
@@ -87,9 +87,21 @@ module.exports = class Client
 				body += data.toString();
 			);
 			response.on("end", ->
-				callback(JSON.parse(body))
+        if body.length
+          parsedBody = {}
+          try
+              parsedBody = JSON.parse(body)
+          catch e
+              console.log(e)
+
+          callback(parsedBody)
+        else
+          callback({})
+
 			);
 		);
 
 		request.write(JSON.stringify(data));
 		request.end();
+
+		return request;
```

Comment:
In this API the error is set inside the error property of the callback. We are currently rewriting the API where we will support promises and will implement the standard `(error, result) => style` callback.